patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch

```diff
@@ -293,7 +293,15 @@
  
                      if (!itemstack.func_190926_b())
                      {
-@@ -2575,6 +2607,40 @@
+@@ -2377,6 +2409,7 @@
+             {
+                 flag = false;
+             }
++            flag = net.minecraftforge.common.ForgeHooks.hasElytraFlightAbility(this,itemstack);
```

Comment:
This doesn't seem to expand the damaging effect to custom Elytra's.
How do you propose to address that?

Reply:
Same way vanilla does it, I'd guess, In the player tick, check if the player's flying and damage your item, use energy, burn mana or whatever.

Reply:
The point I was making was that 'the way vanilla does this' is literally 4 lines above this. And should probably get a hook.

Reply:
Vanilla just hardcodes this to damage the chest item, that code must obviously be bypassed, otherwise every kind of elytra flight would damage the chest, no matter what is in the chest or what provides elytra flight. Also it hardcodes it to 1 durability loss every 20 ticks.

 Mods can provide elytra flight by anything possible, from potions to other item slots. Damaging should be up to the mod how to deal with it, like draining power or durability or whatever. 

With elytra flight provided by entity attribute modifiers there isn't even a reference to the item that provided it anymore, also it does not need to be added by an item and multiple items can provide it. I think the best solution is to let every mod deal with that itself, it can easily be done with tickhandlers.

vanilla armor slot items should be able to use "Item.onArmorTick", so not even needing an own tickhandler there.